crates/wallet/examples/descriptor_with_plan.rs

```diff
@@ -0,0 +1,479 @@
+    // to spend on, we would define and add multiple assets.
+    let assets = Assets::new().add(emergency_key_asset);
+
+    // Automatically generate a plan for spending the descriptor
+    let emergency_plan = descriptor
+        .clone()
+        .plan(&assets)
+        .expect("couldn't create emergency plan");
+
+    // Create an input where we can put the plan data
+    // Add the witness_utxo from the deposit transaction to the input
+    let mut input = psbt::Input {
+        witness_utxo: Some(witness_utxo.clone()),
+        ..Default::default()
+    };
+
+    // Update the input with the generated plan
+    emergency_plan.update_psbt_input(&mut input);
+
+    // Push the input to the PSBT
+    psbt.inputs.push(input);
+
+    // Add a default output to the PSBT
+    psbt.outputs.push(psbt::Output::default());
+
+    // 6. Sign the spending psbt
+
+    // At this point, we have a PSBT that is ready to be signed.
+    // It contains public data in its inputs, and data which needs to be signed
+    // in its `unsigned_tx.{input, output}s`
+
+    // Set up signing infrastructure and sign
+    let secp = Secp256k1::new();
+    let mut sighash_cache = SighashCache::new(&psbt.unsigned_tx);
+    let msg = psbt
+        .sighash_msg(0, &mut sighash_cache, None)
+        .unwrap()
+        .to_secp_msg();
+
+    let emergency_key = Xpriv::from_str(emergency_tprv)
+        .expect("couldn't create emergency key")
+        .private_key;
+    let emergency_sig = secp.sign_ecdsa(&msg, &emergency_key);
+
+    // Insert the signatures into our psbt
+    psbt.inputs[0].partial_sigs.insert(
+        bitcoin::PublicKey::from_str(emergency_pk).unwrap(),
+        bitcoin::ecdsa::Signature {
+            signature: emergency_sig,
+            sighash_type: EcdsaSighashType::All,
+        },
+    );
+
+    // 7. Finalize the psbt. At this point, miniscript will check whether the transaction
+    //   satisfies the descriptor, and will notify you if it doesn't.
+
+    psbt.finalize_mut(&secp)
+        .expect("problem finalizing emergency psbt");
+
+    // 8. If desired, extract the transaction from the psbt and broadcast it. We won't do this
+    // here as it saves messing around with faucets, wallets, etc.
+    let _my_emergency_spend_tx = psbt.extract_tx().expect("failed to extract emergency tx");
+
+    // Let's now try the same thing with the unvault transaction. We just need to make a new
+    // plan, sign a new spending psbt, and finalize it.
+
+    // Build a spend transaction the unvault key path
+    let timelock = LockTime::from_height(after).expect("couldn't format locktime");
+    let unvault_spend_transaction = blank_transaction_with(timelock);
+    let mut psbt = psbt_with_unsigned_tx(unvault_spend_transaction);
+
+    // Format an input containing the previous output (we got that already using `get_vout()`)
+    let txin = TxIn {
+        previous_output,
+        sequence: Sequence::ZERO, // you MUST set this if you have a timelock
```

Comment:
This should also work?

```suggestion
        sequence: Sequence::ENABLE_RBF_NO_LOCKTIME, // disables relative timelock 
```

Reply:
I've added this locally